deps/wazuh_testing/wazuh_testing/modules/fim/classes.py

```diff
@@ -76,6 +78,136 @@ def get_required_attributes(check_attributes, result=None):
+
+            old_attributes = event['data']['old_attributes'].keys() - {'type', 'checksum'}
+            old_intersection = old_attributes ^ required_attributes
+            old_intersection_debug = "Event attributes are: " + str(old_attributes)
+            old_intersection_debug += "\nRequired Attributes are: " + str(required_attributes)
+            old_intersection_debug += "\nIntersection is: " + str(old_intersection)
+
+            assert (old_intersection == set()), (f'Old_attributes and required_attributes are not the same. ' +
+                                                 old_intersection_debug)
+
+
+def validate_registry_value_event(event, checks=None, mode=None):
+    """Check if event is properly formatted according to some checks.
+
+    Args:
+        event (dict): represents an event generated by syscheckd.
+        checks (:obj:`set`, optional): set of XML CHECK_* options. Default `{CHECK_ALL}`
+        mode (:obj:`str`, optional): represents the FIM mode expected for the event to validate.
+    """
+
+    def get_required_attributes(check_attributes, result=None):
+        result = set() if result is None else result
+        for check in check_attributes:
+            mapped = REQUIRED_REG_VALUE_ATTRIBUTES[check]
+
+            if isinstance(mapped, str):
+                result |= {mapped}
+            elif isinstance(mapped, list):
+                result |= set(mapped)
+            elif isinstance(mapped, set):
+                result |= get_required_attributes(mapped, result=result)
+
+        return result
+
+    json_file = 'syscheck_event_windows.json' if sys.platform == "win32" else 'syscheck_event.json'
+    with open(os.path.join(_data_path, json_file), 'r') as f:
+        schema = json.load(f)
+
+    validate(schema=schema, instance=event)
+
+    # Check FIM mode
+    mode = global_parameters.current_configuration['metadata']['fim_mode'] if mode is None else mode.replace('-', '')
+    assert (event['data']['mode']).replace('-', '') == mode, f"The event's FIM mode was '{event['data']['mode']}' \
+        but was expected to be '{mode}'"
+
+    # Check attributes
+    if checks:
+        attributes = event['data']['attributes'].keys() - {'type', 'checksum'}
+
+        required_attributes = get_required_attributes(checks)
+
+        intersection = attributes ^ required_attributes
+        intersection_debug = "Event attributes are: " + str(attributes)
+        intersection_debug += "\nRequired Attributes are: " + str(required_attributes)
+        intersection_debug += "\nIntersection is: " + str(intersection)
+
+        assert (intersection == set()), f'Attributes and required_attributes are not the same. ' + intersection_debug
+
+        # Check add file event
+        if event['data']['type'] == 'added':
+            assert 'old_attributes' not in event['data'] and 'changed_attributes' not in event['data']
+
+        # Check modify file event
+        if event['data']['type'] == 'modified':
```

Comment:
The checks for `added` and `modified` are in place to make sure the data for the file is correct:
- added event:  no `old_attributes` or `changed_attributes` present
- modified event: verifies that the `new attributes` are different to the `old_attributes`

In the case of deletion, that information (attributes) is no longer present, since the it was deleted. 

This are basic checks. Any specific check that is needed, for any kind of event, can be done through a custom validator.